pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -265,37 +274,38 @@ func (m *smcpv2mutator) GetProfiles() []string {
 	return m.smcp.Spec.Profiles
 }
 
-func (m *smcpv2mutator) IsOpenShiftRouteEnabled() *bool {
+func (m *smcpv2mutator) DisableIORIfNotSet() {
 	gateways := m.smcp.Spec.Gateways
+	if gateways != nil && gateways.OpenShiftRoute != nil {
+		return
+	}
 
 	if gateways == nil {
-		return nil
+		gateways = &v2.GatewaysConfig{}
 	}
-
-	route := gateways.OpenShiftRoute
-
-	if route == nil {
-		return nil
+	if gateways.OpenShiftRoute == nil {
+		gateways.OpenShiftRoute = disableIOR
 	}
-
-	return route.Enabled
+	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/gateways", *gateways))
 }
 
-func (m *smcpv2mutator) SetOpenShiftRouteEnabled(value bool) {
-	gateways := m.smcp.Spec.Gateways
-
-	if gateways == nil {
-		gateways = &v2.GatewaysConfig{}
+func (m *smcpv2mutator) EnableGatewayAPIIfNotSet() {
+	isGatewayAPISet := func(spec v2.ControlPlaneSpec) bool {
+		if spec.TechPreview != nil {
+			// ignore error - if techPreview.gatewayAPI has wrong type, we still need to check PILOT_ENABLE_GATEWAY_API
```

Comment:
hmm you say we need to check it in the comment, but then you don't actually check it - which is fine, I don't think we need to. But we should adjust the comment I guess

Reply:
Yes, I forgot to remove that comment.